src/contracts/signer-identity.spec.ts

```diff
@@ -0,0 +1,364 @@
+            contract,
+            'fireStringEvent',
+            {
+              from: signer.activeIdentity,
+              event: {
+                target: 'TestContractEvent',
+                eventName: 'StringEvent',
+              },
+              getEventResult: (_, args) => args.text,
+            },
+            randomString,
+          );
+          expect(eventValue).to.eq(randomString);
+        };
+        await expect(Promise.all([...Array(10)].map(() => runOneTest())))
+          .not.to.be.rejected;
+      });
+    });
+
+    describe('when sending funds', () => {
+      it('can send funds', async () => {
+        const amountToSend = Math.floor(Math.random() * 1e3);
+        const balanceBefore = new BigNumber(await web3.eth.getBalance(accounts[1]));
+        await executor.executeSend(
+          { from: signer.activeIdentity, to: accounts[1], gas: 100e3, value: amountToSend });
+        const balanceAfter = new BigNumber(await web3.eth.getBalance(accounts[1]));
+        const diff = balanceAfter.minus(balanceBefore);
+        expect(diff.eq(new BigNumber(amountToSend))).to.be.true;
+      });
+
+      it('should reject transfer and balance of identity should remain same', async () => {
+        const randomString = Math.floor(Math.random() * 1e12).toString(36);
+        const contract = await executor.createContract(
+          'TestContract', [randomString], { from: signer.activeIdentity, gas: 1e6 });
+        expect(contract).to.be.a('Object');
+
+        const amountToSend = Math.floor(Math.random() * 1e3);
+        const balanceBefore = new BigNumber(await web3.eth.getBalance(signer.activeIdentity));
+        await expect( executor.executeSend({
+          from: signer.activeIdentity,
+          to: contract.options.address,
+          gas: 100e3,
+          value: amountToSend,
+        })).to.rejected;
+
+        const balanceAfter = new BigNumber(await web3.eth.getBalance(signer.activeIdentity));
+        expect(balanceAfter.eq(balanceBefore)).to.be.true;
+      });
+
+      it('should reject transfer and funds should deduct transfer fee in underlaying account',
+        async () => {
+          const randomString = Math.floor(Math.random() * 1e12).toString(36);
+          const contract = await executor.createContract(
+            'TestContract', [randomString], { from: signer.activeIdentity, gas: 1e6 });
+          expect(contract).to.be.a('Object');
+
+          const amountToSend = Math.floor(Math.random() * 1e3);
+
+          const balanceBefore = new BigNumber(await web3.eth.getBalance(signer.underlyingAccount));
+
+          await expect( executor.executeSend({
+            from: signer.activeIdentity,
+            to: contract.options.address,
+            gas: 100e3,
+            value: amountToSend,
+          })).to.rejected;
+
+          const balanceAfter = new BigNumber(await web3.eth.getBalance(signer.underlyingAccount));
+          expect(balanceAfter.eq(balanceBefore)).to.be.not.true;
```

Comment:
not sure about this code, but looks better to me than testing for .eq to be not true ;) 

```suggestion
          expect(balanceAfter).to.be.lt(balanceBefore);
```

Reply:
check also this style for the other tests

Reply:
ah I see, you implemented it like this, bc of BigNumber?

Reply:
Updated tests to use `.lt` and `.gt`, currently there's no real need for chai-bignumber, I think ;)